test/mod/test_util_path.py

```diff
@@ -18,20 +19,45 @@ def tempdir_fixture():
 
 
 def test_clamp_mtime(tempdir):
-    start = int(time.time())
```

Comment:
I wonder if we could just move the "start" of the clamp a little bit earlier. The test does not care and that would also protect us from random clock jumps without the need of a loop or a sleep(). What do you think about something like (diff against the original one)?
```python
diff --git a/test/mod/test_util_path.py b/test/mod/test_util_path.py
index cfeb1656..46c15715 100644
--- a/test/mod/test_util_path.py
+++ b/test/mod/test_util_path.py
@@ -18,7 +18,10 @@ def tempdir_fixture():
 
 
 def test_clamp_mtime(tempdir):
-    start = int(time.time())
+    # this should not be precise, anything created within the last 60min
+    # should be "clamped" and by adding a bit of slack chrony and random
+    # RTC jumps will not affect the test
+    clamp_start = int(time.time() - 60*60)
 
     tree = Path(tempdir, "tree")
     tree.mkdir()
@@ -33,12 +36,14 @@ def test_clamp_mtime(tempdir):
     link = Path(tree, "link")
     link.symlink_to(folder, target_is_directory=True)
 
-    timepoint = 1644758228
-    path.clamp_mtime(tree, start, timepoint)
+    # >>> datetime.datetime.fromtimestamp(1644758228)
+    # datetime.datetime(2022, 2, 13, 14, 17, 8)
+    clamp_to = 1644758228
+    path.clamp_mtime(tree, clamp_start, clamp_to)
 
     def ensure_mtime(target, dfd):
         stat = os.stat(target, dir_fd=dfd, follow_symlinks=False)
-        assert stat.st_mtime <= timepoint, f"failed for {target} ({dfd})"
+        assert stat.st_mtime <= clamp_to, f"failed for {target} ({dfd})"
 
     for _, dirs, files, dfd in os.fwalk(tree):
         ensure_mtime(".", dfd)
```
(numbers are a bit arbitrary). I also changed some of the vars to make it a bit more obvious what is going on, I found it a little bit hard to follow (but maybe subjective)

Reply:
I like it when we can avoid _sleeping_ during code execution in general 😅
I'd really go for 61min or take 120 as it wouldn't matter I guess - I'll update…

Reply:
Thanks! Yeah, please tweak as you see fit, I have no strong opinion either way.